synapse/http/site.py

```diff
@@ -500,6 +501,23 @@ def _should_log_request(self) -> bool:
 
         return True
 
+    def get_client_ip_if_available(self) -> str:
+        """Logging helper. Return something useful when a client IP is not retrievable
+        from a unix socket.
+
+        In practice, this returns the socket file path on a SynapseRequest if using a
+        unix socket and the normal IP address for TCP sockets.
+
+        """
+        # If using a unix socket, getClientAddress() is supposed to return a
+        # UNIXAddress object containing the path and file of the socket, but for some
+        # reason it's 'None' which isn't helpful. getPeer() also doesn't exist, however
+        # getHost() does exist and reports the socket file correctly.
+        if isinstance(self.getClientAddress(), UNIXAddress):
```

Comment:
should probably extract `client_address = self.getClientAddress()` just to guard against anything silly where it might return two different things.

But.. `isinstance(..., UNIXAddress)` seems to contradict your comment where it says it returns `None`; I don't see `None` being an instance of `UNIXAddress`?


We also probably don't want to log the path of the unix socket in the first place — I doubt it'd be useful for the logs? I thought you could get the PID of the calling process over a unix socket but even if you can't, I'd probably just stub it out...

Reply:
I took great care that it would return what it was supposed to at the correct moment. There are four combinations it occurs: `SynapseRequest`, `XForwardedForRequest` for both TCP and UNIX connections. While this function is attached to `SynapseRequest`, it is brought into `XForwardedForRequest` as the former is the base class.

For TCP:
* it returns an IP address(pre-existing behavior)

For UNIX:
* It will say the `/path/to/socket` for a `SynapseRequest`(e.g. /health, or a replication connection), or
* an IP address for an `XForwardedForRequest`

I'm thinking change that comment line from `reason it's 'None'` to `reason it contains 'None'` since it's still a UNIXAddress class, just the value it contains is 'None'. Does that make sense?

Considering it's in the *logs* and the public generally doesn't get to see those, I don't think it matters really...But could just make it say something like `Unix Socket Request`, as it isn't functionally required to complete a real request.

Example log line:
```log
synapse | 2023-03-29 09:34:00,817 - synapse.access.http.unix - 461 - INFO - POST-2 - 172.17.44.1 - unix - {None} Processed request: 0.297sec/0.001sec (0.044sec, 0.008sec) (0.006sec/0.116sec/13) 149B 200 "POST /_matrix/client/v3/register HTTP/1.0" "Go-http-client/1.1" [0 dbevts]
```

Reply:
Updated comment to clarify too